kolibri/plugins/coach/assets/src/vue/groups-page/rename-group-modal.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+
+  <core-modal :title="$tr('renameLearnerGroup')"
+    @cancel="close">
+    <div>
+      <form @submit.prevent="callRenameGroup">
+        <textbox type="text"
+          :label="$tr('learnerGroupName')"
+          :aria-label="$tr('learnerGroupName')"
+          :autofocus="true"
+          :required="true"
+          :invalid="invalid"
+          v-model.trim="groupNameInput" />
+        <icon-button :text="$tr('cancel')"
+          @click="close" />
+        <icon-button :text="$tr('save')"
+          :primary="true"
+          type="submit" />
+      </form>
+    </div>
+  </core-modal>
+
+</template>
+
+
+<script>
+
+  const groupActions = require('../../group-actions');
+
+  module.exports = {
+    $trNameSpace: 'confirm-enrollment-modal',
+    $trs: {
+      renameLearnerGroup: 'Rename Learner Group',
+      learnerGroupName: 'Learner Group Name',
+      cancel: 'Cancel',
+      save: 'Save',
+    },
+    data() {
+      return {
+        groupNameInput: this.groupName,
+        invalid: false,
+      };
+    },
+    components: {
+      'core-modal': require('kolibri.coreVue.components.coreModal'),
+      'textbox': require('kolibri.coreVue.components.textbox'),
+      'icon-button': require('kolibri.coreVue.components.iconButton'),
+    },
+    props: {
+      className: {
+        type: String,
+        required: true,
+      },
+      classId: {
+        type: String,
+        required: true,
+      },
+      groupName: {
+        type: String,
+        required: true,
+      },
+      groupId: {
+        type: String,
+        required: true,
+      },
+    },
+    methods: {
+      callRenameGroup() {
+        this.renameGroup(this.classId, this.groupId, this.groupNameInput);
```

Comment:
I don't know if this is better, but since it's a thin wrapper, might be worth mentioning:

Couldn't you just call this from the view?

```
<form @submit.prevent="renameGroup(classId, groupId, groupNameInput)">
     <textbox type="text"
       :label="$tr('learnerGroupName')"
       :aria-label="$tr('learnerGroupName')"
       :autofocus="true"
       :required="true"
       :invalid="invalid"
       v-model.trim="groupNameInput" />
     <icon-button :text="$tr('cancel')"
       @click="close" />
     <icon-button :text="$tr('save')"
       :primary="true"
       type="submit" />
</form>
```